### PR TITLE
cln: add "texinfo" for head

### PR DESCRIPTION
**Pull request by Fenyutanchan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
According to Homebrew/discussions#4372 and Homebrew/discussions#4218.
Thanks for @carlocab and @cho-m ~

### CODE REVIEW

File: Formula/cln.rb [hunk 1]
```diff
@@ -35,6 +35,10 @@ class Cln < Formula
     depends_on "autoconf" => :build
     depends_on "automake" => :build
     depends_on "wget" => :build
+
+    on_system :linux, :macos_ventura_or_newer do
```
Comment (ZhongRuoyu) [inline Formula/cln.rb:39]:
```suggestion
    on_system :linux, macos: :ventura_or_newer do
```

**Comment by ZhongRuoyu** [comment]
Can you please format your commit according to our [commit style](https://docs.brew.sh/Formula-Cookbook#commit)? Thanks!

File: Formula/cln.rb (hunk 1 above)
Comment (Fenyutanchan) [reply Formula/cln.rb]: Sorry for that.
I was checking the guidelines before and I have pushed all commits with the flag `-f`.
Finally, these modifications are accepted.

**Comment by Fenyutanchan** [comment]
Thank you for all.
I'm not familiar with the contributing rules here.
@carlocab and @ZhongRuoyu.

File: Formula/cln.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class Cln < Formula
   desc "Class Library for Numbers"
   homepage "https://www.ginac.de/CLN/"
-  license "GPL-2.0"
+  license "GPL-2.0-only"
```
Comment (ZhongRuoyu) [inline Formula/cln.rb:4]: This should be `GPL-2.0-or-newer`. See: https://www.ginac.de/CLN/cln.git/?p=cln.git;a=blob;f=COPYING;h=20774a1d47c8194ae5c09b589094735c0e977772;hb=HEAD#l300

```suggestion
  license "GPL-2.0-or-newer"
```

Comment (Fenyutanchan) [reply Formula/cln.rb]: Yeah, I omitted it.
Thank you~

**Comment by ZhongRuoyu** [comment]
Thanks @Fenyutanchan. I'm happy once you squash all the commits into one.

**Comment by Fenyutanchan** [comment]
> 

Done it.

**Review by ZhongRuoyu — APPROVED** [review]
Thank you!

**Pull request merged**